backpack/core/derivatives/lstm.py

```diff
@@ -0,0 +1,348 @@
+            ifgo[t, :, H3:H4] = sigmoid(ifgo[t, :, H3:H4])
+            c[t] = ifgo[t, :, H0:H1] * ifgo[t, :, H2:H3]
+            if t != 0:
+                c[t] += ifgo[t, :, H1:H2] * c[t - 1]
+            c_tanh[t] = tanh(c[t])
+            h[t] = ifgo[t, :, H3:H4] * c_tanh[t]
+
+        return ifgo, c, c_tanh, h
+
+    @classmethod
+    def _ifgo_jac_t_mat_prod(cls, module: LSTM, mat: Tensor) -> Tensor:
+        V: int = mat.shape[0]
+        T: int = mat.shape[1]
+        N: int = mat.shape[2]
+        H: int = module.hidden_size
+        H0: int = 0 * H
+        H1: int = 1 * H
+        H2: int = 2 * H
+        H3: int = 3 * H
+        H4: int = 4 * H
+
+        ifgo, c, c_tanh, _ = cls._forward_pass(module, mat)
+
+        # backward pass
+        H_prod_t: Tensor = zeros(V, N, H, device=mat.device, dtype=mat.dtype)
+        C_prod_t: Tensor = zeros(V, N, H, device=mat.device, dtype=mat.dtype)
+        C_prod_old: Tensor = zeros(V, N, H, device=mat.device, dtype=mat.dtype)
+        IFGO_prod: Tensor = zeros(V, T, N, 4 * H, device=mat.device, dtype=mat.dtype)
+        for t in reversed(range(T)):
+            # jac_t_mat_prod until node h
+            H_prod_t[:] = mat[:, t]
+            if t != (T - 1):
+                H_prod_t += einsum(
+                    "vnh,hg->vng",
+                    IFGO_prod[:, t + 1],
+                    module.weight_hh_l0,
+                )
+
+            # C_prod_t = jac_t_mat_prod until node c
+            if t != (T - 1):
+                C_prod_old[:] = C_prod_t
+            C_prod_t[:] = einsum(
+                "vnh,nh->vnh",
+                H_prod_t,
+                ifgo[t, :, H3:H4] * (1 - c_tanh[t] ** 2),
+            )
+            if t != (T - 1):
+                C_prod_t += einsum(
+                    "vnh,nh->vnh",
+                    C_prod_old,
+                    ifgo[t + 1, :, H1:H2],
+                )
+
+            IFGO_prod[:, t, :, H3:H4] = einsum(
+                "vnh,nh->vnh",
+                H_prod_t,
+                c_tanh[t] * (ifgo[t, :, H3:H4] * (1 - ifgo[t, :, H3:H4])),
+            )
+            IFGO_prod[:, t, :, H0:H1] = einsum(
+                "vnh,nh->vnh",
+                C_prod_t,
+                ifgo[t, :, H2:H3] * (ifgo[t, :, H0:H1] * (1 - ifgo[t, :, H0:H1])),
+            )
+            if t >= 1:
+                IFGO_prod[:, t, :, H1:H2] = einsum(
+                    "vnh,nh->vnh",
+                    C_prod_t,
+                    c[t - 1] * (ifgo[t, :, H1:H2] * (1 - ifgo[t, :, H1:H2])),
+                )
+            IFGO_prod[:, t, :, H2:H3] = einsum(
+                "vnh,nh->vnh",
+                C_prod_t,
+                ifgo[t, :, H0:H1] * (1 - ifgo[t, :, H2:H3] ** 2),
+            )
+        return IFGO_prod
+
+    def _jac_mat_prod(
+        self,
+        module: LSTM,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+        mat: Tensor,
+        sum_batch: bool = True,
+    ) -> Tensor:
+        V: int = mat.shape[0]
```

Comment:
@schaefertim Should there be a `self.check_parameters()` here?

Reply:
Yes! One more reason refactor the `check_parameters()`